### ISSUE TITLE
[Feature request] Add option to fetch data from other sources

### ISSUE THREAD

**Issue by ghostbuster91** [SmiteshP/nvim-navic#139]
Hi,

Thanks for the great plugin. 
I wonder if it would be possible to allow getting data from other sources than LSP. 

For example I would like to populate it with data from the treesitter if LSP is not available. 
This would allow me achieving something similar to https://github.com/phelipetls/jsonpath.nvim

**Comment by pidgeon777** [comment]
Yes, it would be great for navic to also access treesitter information.

Also, navbuddy could improve itself by using treesitter as a source.
